### ISSUE TITLE
Cyclic dependency between twisted.test.test_failure and twisted.test.generator_failure_tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4562]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4562|
|Type|defect|
|Created|2010-07-09 18:53:20Z|
|Branch|https://github.com/twisted/twisted/tree/generator-tests-cycle-4562|

test_failure imports generator_failure_tests.  generator_failure_tests imports test_failure.  This works as long as test_failure gets loaded first, but fails as soon as you try to start with generator_failure_tests (which you can observe by picking them specifically with trial: `trial twisted.test.generator_failure_tests`).

The dependency is really shallow though, so it should be easy to fix.

This is something that was discovered in the [#1784](https://github.com/twisted/twisted/issues/1784), and the fix is pulled straight from that branch as well.

<details><summary>Searchable metadata</summary>

```
trac-id__4562 4562
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_generator_tests_cycle_4562 branches/generator-tests-cycle-4562
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1278701600000000 1278701600000000
changetime__1278938891000000 1278938891000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29737]) Branching to 'generator-tests-cycle-4562'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [29756]) Drop this; see [#4562](https://github.com/twisted/twisted/issues/4562).

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess set owner to @exarkun|
|-|-|

Thanks for fixing this, exarkun. Looks good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [29772]) Merge generator-tests-cycle-4562

Author: exarkun
Reviewer: jesstess
Fixes: [#4562](https://github.com/twisted/twisted/issues/4562)

Remove the circular dependency between twisted.test.test_failure
and twisted.test.generator_failure_tests.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
